File: class23_homework/A-Asen-Stoilov-4/Java/src/Elsys/Homework/Plane.java
package Elsys.Homework;

import java.util.Scanner;


public class Plane {

	private int[][] seats;
	private int seatcount;
	private int[] freeseats;
	
	public Plane(){
		this.seats = new int[27][6];
		this.seatcount = 0;
		this.freeseats = new int[27];
		
		for(int i=0; i< 27 ; i++)
			freeseats[i] = 6;
	}
	
	private void findSeat(int count, int size){
		int free;
		if (freeseats[count]==6){
			free=0;
		} else {
			free=6-freeseats[count];
		}
		
		for (int i = free; i<free+size; i++){
			System.out.printf("<Pasenger seated ROW:%d SEAT:%d>\n", count+1, i+1);
			seats[count][i] = 1;
		}
	}
	
	private void addGroup(){
		Group gr = new Group();
		int grsize = gr.getSize();
		for (int i=this.seatcount; i<27; i++){
			if (freeseats[i] >= grsize){
				findSeat(i,grsize);
				freeseats[i] -=grsize;
				break;
			} else{
				if (i==26)
					System.out.println("<There are no free seats for this group>");
			}
		}
		if (freeseats[seatcount] == 0) seatcount+=1;
	}
	
	private void printSeats(){
		for (int i=0;i<27;i++){
			for(int j=0;j<6;j++){
				if(seats[i][j] == 1){
					System.out.printf("<ROW:%d|SEAT:%d - OCUPIED>\n",i+1,j+1);
				} else {
					System.out.printf("<ROW:%d|SEAT:%d - FREE>\n",i+1,j+1);
				}
			}
		}
	}
	
	public void getPlaneReady() {
		Scanner scan = new Scanner(System.in);
		while(this.seatcount<27){
			System.out.println();
			System.out.println("press ENTER...");
			String readString =scan.nextLine();
			if (readString.equals("")){
				addGroup();
				printSeats();	
			}
		}
		scan.close();
	}
	
}

File: class23_homework/A-Asen-Stoilov-4/Ruby/plane.rb
class Plane 
	attr_accessor :seats
	attr_accessor :freeseats
	attr_accessor :seatcount
	attr_accessor :freecount
	
	def initialize
		self.freecount = 26*6
		self.seatcount = 0
		self.seats =  Array.new(26) { Array.new(6, 0) }
		self.freeseats = Array.new(26)
		(0..25).each do |i|
			self.freeseats[i] = 6
		end
	end
	
	def getPlaneReady
	
		while self.freecount > 0
		
			puts 'press the enter key'
			names = gets

			if names == "\n"
				addGroup
				printseats
				puts self.freecount
			end
		end
	end
	
	private
	
	def findseat count,size
		free = 0
		if freeseats[count] < 6
			free = 6 - freeseats[count] 
		end
		(free..(free+size-1)).each do |i|
			puts "<Pasenger seated ROW:#{count+1} SEAT:#{i+1}>\n"
			seats[count][i] = 1
			self.freecount -=1	
		end
	end
	
	def addGroup
		
		grsize = 1 + rand(3)
		(seatcount..25).each do |i|
			if freeseats[i] >= grsize 
				findseat i,grsize
				freeseats[i] -=grsize
				break 		
			elsif i==25
				puts "<Pasengers can't be seated>"
			end
			
		end
		
		if freeseats[seatcount] == 0
				self.seatcount +=1
			end	
	end
	
	def  printseats
		(0..25).each do |i|
			(0..5).each do |j|
				if seats[i][j] == 1
					puts "<ROW:#{i+1}|SEAT:#{j+1} - OCCUPIED>\n"
				else
					puts "<ROW:#{i+1}|SEAT:#{j+1} - FREE>\n"
				end
			end
		end
	end
end

plane = Plane.new
plane.getPlaneReady
